src/app/dependency-injection/using-injector/base-class-for-injector/base-class-for-injector.component.ts

```diff
@@ -0,0 +1,18 @@
+import { Component, Injector, OnInit } from '@angular/core';
+import { LoggerService } from '../../../../shared/logger.service';
+
+@Component({
+  selector: 'app-base-class-for-injector',
+  templateUrl: './base-class-for-injector.component.html',
+})
+export class BaseClassForInjectorComponent implements OnInit {
+  readonly logger: LoggerService;
+
+  constructor(private injector: Injector) {
```

Comment:
don't forget to make your DI dependencies `readonly`, because inside ngOnInit I can do the next:
`this.injector = {}` or `this.injection = undefined`, so our service is not protected to be changed.